### ISSUE TITLE
[Bug]: Asian characters are not rendered properly

### ISSUE THREAD

**Issue by fishfree** [ngandrass/moodle-quiz_archiver#58]
### Quiz archiver moodle plugin version

latest

### Quiz archiver worker version

latest

### Worker service deployment method

Other, please specify below

### Moodle Version

Moodle 4.2

### PHP Version

PHP 7.4

### Database

MySQL

### Operating system

Ubuntu 22.04

### What happened?

Expected behavior:
All characters are normal displayed
Actual behavior:
Chinese / Japanese / Korean characters are gabbled as below:
![image](https://github.com/user-attachments/assets/c990ed55-f3ac-44f2-b4f9-fdc9e1f5345e)

Additional information:

### Steps to reproduce

Just do it normally.

**Comment by ngandrass** [comment]
Thanks for your bug report. The quiz archive worker always uses the font of your Moodle / theme.

Usually, these fonts are directly served by your Moodle instance and therefore should be present. However, there seems to be a problem with the asian characters and I was able to reproduce it.

![image](https://github.com/user-attachments/assets/24eb99ba-9515-4154-8b88-ae8632ae7e87)

I'll look into this problem and let you know once there is a fix :)

**Comment by ngandrass** [comment]
Hi @fishfree,

I did some testing and found that Moodle actually does not serve the basic font files. Since you are not using the Docker container but deploying the quiz archive worker manually, you must make sure that the full fonts are present on your sever.

**If you are on Debian / Ubuntu, installing the package `fonts-noto` should be sufficient.**

If this fixes the problem for you please shortly report back here. I'll then update the Docker container and the installation documentation for manual deployments accordingly.

Thanks!

**Comment by fishfree** [comment]
@ngandrass I see. I used your demo free worker service. I expierenced a similar problem. After installing fonts-noto, it disappeared. Many thanks!

**Comment by ngandrass** [comment]
Yeah, the current Docker image is actually still missing the extended font sets. I wanted to confirm that this change really solves your problem before I release a new version.

I'll update the Docker image and add a note to the documentation. After this release, the demo worker service will also work with the asian characters :)

Thanks again for your bug report.

**Comment by ngandrass** [comment]
## For future readers

The problem has been fixed with the release of [Archive Worker v2.1.3](https://github.com/ngandrass/moodle-quiz-archive-worker/releases/tag/v2.1.3).

If you are using the official Docker image, make sure to use at least [version 2.1.3](https://hub.docker.com/layers/ngandrass/moodle-quiz-archive-worker/2.1.3/images/sha256-00b5a7aedd75d0353afccccf75129f7fcd2c3307ad14ca46102508438d5bcb5e).

I also added a note for manual deployments in the [pitfalls sections of the official documentation](https://quizarchiver.gandrass.de/configuration/initialconfig/pitfalls/#text-is-not-rendered-correctly)

**Comment by fishfree** [comment]
@ngandrass Hello, I find this feature not working again in the latest docker image.

**Comment by ngandrass** [comment]
@fishfree I'm sorry to hear that this happens again. I can reproduce the problem and will provide a fix today. Must be a regression from the base image update and size optimizations ...

**Comment by ngandrass** [comment]
Sorry again for the regression. The new worker release [v3.3.2](https://github.com/ngandrass/moodle-quiz-archive-worker/releases/tag/v3.3.2) fixes this issue. Updating your current Docker container to the newest image will solve this problem for you.

The problem was, that the required fonts are actually not directly in the `fonts-noto` package, but in `fonts-noto-cjk` instead. This package was only marked as a "recommended install" which I disabled previously for optimizing the image size.

To prevent this problem from re-appearing in the future, I added the required packages explicitly to the list of dependencies to install. Therefore they will always be installed :)
